Add support for tasks and unordered list markdown

imf-ali opened the pull request:
## Issue(s)	
Tasks and unordered list does not appear in correct format as that in RC web app.

## How to test or reproduce
- [x] Tasks should display in expected form
- [x] Unordered list should display in expected form

## Screenshots
Previous behaviour
<img width="267" alt="image" src="https://user-images.githubusercontent.com/42451210/228930970-80c2f57d-a014-49a3-833c-49e4f9b2a790.png">

Improvement
<img width="294" alt="image" src="https://user-images.githubusercontent.com/42451210/228929777-7eac48ac-189a-4c68-afc8-b9ff7f6a778f.png">


## Types of changes
<!-- What types of changes does your code introduce to Rocket.Chat? -->
<!-- Put an `x` in the boxes that apply -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist
<!-- Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code. -->

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

Fix #4927

imf-ali commented:
Hey @diegolmello and @GleidsonDaniel , please have a look at the PR.

imf-ali commented on app/containers/markdown/new/TaskList.tsx:
```diff
@@ -16,7 +17,9 @@ const TaskList = ({ value = [] }: ITasksProps) => {
 				<View style={styles.row}>
-					<Text style={[styles.text, { color: colors.bodyText }]}>{item.status ? '- [x] ' : '- [ ] '}</Text>
+					<Text style={[styles.text, { color: colors.bodyText }]}>
+						<CheckBox disabled={true} style={{ height: 20, width: 20, margin: 5 }} boxType='square' value={item.status} />
```
Done.

imf-ali commented:
@GleidsonDaniel please have a look at it.

GleidsonDaniel commented:
> A quick comment, instead add a new library to the repo, can you use it the same way as other places?
> 
> As you can see here:
> 
> https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/2d07b1682ca8ca06eb1addfc8b6bbd3287f220ae/app/containers/MessageBox/index.tsx#L1119-L1124
> 
> we are using the custom icon for the checkbox

@imf-ali i think this is the best approach.
Change your PR following this pattern commented by @reinaldonetof

GleidsonDaniel reviewed (changes requested):
Change your PR following the pattern commented by @reinaldonetof

imf-ali commented:
Changes done. Please have a look @GleidsonDaniel @reinaldonetof .
Preview of the change
<img width="310" alt="image" src="https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/42451210/6a9cc510-0364-4549-accc-78cab7515133">

GleidsonDaniel commented on ios/Podfile.lock:
```diff
@@ -505,6 +506,9 @@ PODS:
     - React-Core
   - RNCAsyncStorage (1.17.11):
     - React-Core
+  - RNCCheckbox (0.5.15):
```
update your podfile

imf-ali replied:
Done

imf-ali commented:
@GleidsonDaniel comment addressed. Please have a look.

GleidsonDaniel commented on ios/Podfile.lock:
```diff
@@ -997,4 +997,4 @@ SPEC CHECKSUMS:
 PODFILE CHECKSUM: da36ddb728f8698fcf3e8b17d853ccedc4f8a24c
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.11.3
```
```suggestion
COCOAPODS: 1.12.1
```

imf-ali replied:
Done.

GleidsonDaniel commented on app/containers/markdown/new/TaskList.tsx:
```diff
@@ -16,7 +17,14 @@ const TaskList = ({ value = [] }: ITasksProps) => {
+							testID={item.status ? 'send-to-channel-checked' : 'send-to-channel-unchecked'}
+							name={item.status ? 'checkbox-checked' : 'checkbox-unchecked'}
+							size={24}
+							color='#9297a2'
```
Don't put the color directly in the code, use a constant, preferably one that is in the app's const colors, as it doesn't change color, add it as a fixed color.

GleidsonDaniel commented on app/containers/markdown/new/TaskList.tsx:
```diff
@@ -16,7 +17,14 @@ const TaskList = ({ value = [] }: ITasksProps) => {
-					<Text style={[styles.text, { color: colors.bodyText }]}>{item.status ? '- [x] ' : '- [ ] '}</Text>
+					<Text style={[styles.text, { color: colors.bodyText }]}>
+						<CustomIcon
+							testID={item.status ? 'send-to-channel-checked' : 'send-to-channel-unchecked'}
```
```suggestion
							testID={item.status ? 'task-list-checked' : 'task-list-unchecked'}
```

imf-ali commented:
> I saw that you sent me a private message asking for the review. Just pay attention to one issue: Previously your PR was not reviewed because it didn't even pass the lint, so before requesting the PR check that all the steps in the checklist are correct.

Oh, sorry for the inconvenience. Will make sure next time, all the checks are successful. I have addressed the remaining comments. Kindly take a look once.
